src/main/java/com/bunq/sdk/http/ApiClient.java

```diff
@@ -106,163 +92,207 @@
+  public static final int OK_STATUS_CODE = 200;
 
   /**
    * Private variables.
    */
-  private CloseableHttpClient httpClient;
+  private OkHttpClient httpClient;
   private ApiContext apiContext;
 
   /**
    * @param apiContext API context to make the calls in.
    */
   public ApiClient(ApiContext apiContext) {
     this.apiContext = apiContext;
-    this.httpClient = buildHttpClient();
+    this.httpClient = buildOkHttpClient();
   }
 
-  private CloseableHttpClient buildHttpClient() {
-    try {
-      // TODO: Fix SSL handling
-      SSLContextBuilder builder = new SSLContextBuilder();
-      SSLConnectionSocketFactory sslConnectionSocketFactory =
-          new SSLConnectionSocketFactory(builder.build());
-      HttpClientBuilder httpClientBuilder = HttpClients
-          .custom()
-          .setSSLSocketFactory(sslConnectionSocketFactory);
-      setProxyIfNeeded(httpClientBuilder);
-
-      return httpClientBuilder.build();
-    } catch (NoSuchAlgorithmException | KeyManagementException | MalformedURLException exception) {
-      throw new UncaughtExceptionError(exception);
-    }
+  private OkHttpClient buildOkHttpClient() {
+    OkHttpClient.Builder clientBuilder;
+
+    clientBuilder = new OkHttpClient().newBuilder()
+        .certificatePinner(determineCertificateToPin())
+        .connectTimeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+        .readTimeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+        .writeTimeout(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+    setProxyIfNeeded(clientBuilder);
+
+    return clientBuilder.build();
   }
 
-  private void setProxyIfNeeded(HttpClientBuilder httpClientBuilder)
-      throws MalformedURLException {
+  private void setProxyIfNeeded(OkHttpClient.Builder httpClientBuilder) {
     String proxyString = apiContext.getProxy();
 
     if (proxyString != null) {
-      Uri proxyUri = new Uri(proxyString);
-      HttpHost proxy = new HttpHost(proxyUri.getHost(), proxyUri.getPort(), proxyUri.getScheme());
-      httpClientBuilder.setProxy(proxy);
+      URL url = Objects.requireNonNull(HttpUrl.parse(proxyString)).url();
+      Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(url.getHost(), url.getPort()));
+      httpClientBuilder.proxy(proxy);
     }
   }
 
+  private CertificatePinner determineCertificateToPin() {
+    CertificatePinner.Builder certificateBuilder = new CertificatePinner.Builder();
+
+    switch (apiContext.getEnvironmentType()) {
+      case PRODUCTION:
+        return certificateBuilder.add(
+            apiContext.getBaseUri(), getPinnedKeyStringProduction()
+        ).build();
+      case SANDBOX:
+        return certificateBuilder.add(
+            apiContext.getBaseUri(), getPinnedKeyStringSandbox()
+        ).build();
+      default:
+        throw new BunqException(
+            String.format(ERROR_AMI_ENVIRONMENT_NOT_EXPECTED, apiContext.getEnvironmentType().toString())
+        );
+    }
+  }
+
+  private String getPinnedKeyStringSandbox() {
```

Comment:
Feels a bit weird to use a method to get a constant, maybe it's just me but wouldn't it be nicer to just access the constant directly? As it already's really clear what the constants are as the names are ok.